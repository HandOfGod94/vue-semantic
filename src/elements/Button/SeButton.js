export default {
  name: 'se-button',
  props: {
    as: { type: String, default: 'button' },
    active: Boolean,
    attach: String,
    color: String,
    label: String,
    labelPosition: { type: String, default: '' },
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  },
  computed: {
    classObject () {
      return {
        fluid: this.fluid,
        [this.size]: this.size,
        compact: this.compact,
        circular: this.circular,
        ui: true, // Semantic ui class
        [`${this.attach} attached`]: this.attach,
        [`${this.float} floated`]: this.float,
        disabled: this.disabled,
        inverted: this.inverted,
        [this.color]: this.color,
        loading: this.loading,
        [`${this.labelPosition} labeled`]: this.labelPosition || this.label,
        icon: this.icon,
        basic: this.basic,
        button: true
      }
    }
  },

  render (h) {
    let Component = this.as
    return (
      <Component class={this.classObject}>
        {this.icon && <i class={`${this.icon} icon`}></i> }
        {this.$slots.label || this.label}
        {this.$slots.default}
      </Component>
    )
  }
}
