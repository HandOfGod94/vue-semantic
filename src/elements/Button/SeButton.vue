<template>
  <component :is="as" :class="classObject">
    <i :class="`${this.icon} icon`" v-if="classObject.icon"></i>
    <slot name="label">
      {{label}}
    </slot>
    <slot></slot>
  </component>
</template>

<script>
export default {
  name: 'se-button',
  props: {
    as: { type: String, default: 'button' },
    active: Boolean,
    attach: String,
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  },
  computed: {
    classObject () {
      return {
        [this.size]: this.size,
        compact: this.compact,
        circular: this.circular,
        ui: true, // Semantic ui class
        [`${this.attach} attached`]: this.attach,
        [`${this.float} floated`]: this.float,
        disabled: this.disabled,
        inverted: this.inverted,
        [this.color]: this.color,
        loading: this.loading,
        labeled: this.label,
        icon: this.icon,
        basic: this.basic,
        button: true
      }
    }
  }
}
</script>
