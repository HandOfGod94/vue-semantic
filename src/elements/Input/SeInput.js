import SeInputComponent from './SeInput.functional'

export default {
  name: 'se-input',
  props: {
    type: { type: String, required: true },
    action: Boolean,
    actionPosition: String,
    disabled: Boolean,
    fluid: Boolean,
    placeholderText: String,
    focus: Boolean,
    loading: Boolean,
    icon: String,
    iconPosition: String,
    labelPosition: String,
    error: Boolean,
    value: String,
    name: String
  },

  computed: {
    classObject () {
      return {
        ui: true,
        fluid: this.fluid,
        disabled: this.disabled,
        [`${this.actionPosition} action`]: this.action,
        [`${this.labelPosition} labeled`]: this.labelPosition,
        focus: this.focus,
        [this.iconPosition]: this.iconPosition,
        icon: this.icon,
        loading: this.loading,
        input: true,
        error: this.error
      }
    },
    inputListener () {
      const inputEvent = { input: event => this.$emit('input', event.target.value) }
      return Object.assign({}, this.$listeners, inputEvent)
    }
  },

  render (createElement) {
    return createElement(SeInputComponent, {
      attrs: {
        ...this.$props,
        slots: this.$slots
      },
      class: this.classObject,
      on: this.inputListener
    })
  }
}
