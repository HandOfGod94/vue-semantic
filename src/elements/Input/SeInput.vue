<template>
  <div :class="classObject">
    <slot name="left-action"></slot>
    <slot name="left-label"></slot>
    <input :type="type" :placeholder="placeholderText" :value="value" v-on="inputListener" />
    <i v-if="icon" :class="`${classObject.icon} icon`"></i>
    <slot name="right-label"></slot>
    <slot name="right-action"></slot>
  </div>
</template>

<script>
import SeLabel from '../Label/SeLabel.vue'

export default {
  name: 'se-input',
  props: {
    type: { type: String, required: true },
    action: Boolean,
    actionPosition: String,
    disabled: Boolean,
    placeholderText: String,
    focus: Boolean,
    loading: Boolean,
    icon: String,
    iconPosition: String,
    labelPosition: String,
    error: Boolean,
    value: String
  },
  computed: {
    classObject () {
      return {
        ui: true,
        disabled: this.disabled,
        [`${this.actionPosition} action`]: this.action,
        [`${this.labelPosition} labeled`]: this.labelPosition,
        focus: this.focus,
        [this.iconPosition]: this.iconPosition,
        icon: this.icon,
        loading: this.loading,
        input: true,
        error: this.error
      }
    },
    inputListener () {
      let vm = this
      let inputEvent = {
        input: function (event) {
          vm.$emit('input', event.target.value)
        }
      }
      return Object.assign({}, this.$listeners, inputEvent)
    }
  },
  components: {
    SeLabel
  }
}
</script>
