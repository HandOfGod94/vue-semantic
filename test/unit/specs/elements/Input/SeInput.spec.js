import SeInput from '@/elements/Input/SeInput'
import SeButton from '@/elements/Button/SeButton'
import { mount } from '@vue/test-utils'
import Vue from 'vue'

const actionSlot = `
  <template>
    <se-button color="teal" icon="cart" label="Checkout" />
  </template>
`

const ParentComponent = {
  template: `
    <div id="test">
      <se-input type="text" name="test" v-model="testdata" />
    </div>
  `,
  data () {
    return {
      testdata: 'worldhello'
    }
  }
}

describe('SeInput props testing', () => {
  it('should render basic text input field', () => {
    const data = {
      propsData: { type: 'text' }
    }
    const wrapper = mount(SeInput, data)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have focus on the filed', () => {
    const data = {
      propsData: { type: 'text', focus: true }
    }
    const wrapper = mount(SeInput, data)
    expect(wrapper.classes()).toContain('focus')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have left action button', () => {
    const data = {
      propsData: { type: 'text', action: true, actionPosition: 'left' },
      slots: { leftAction: actionSlot },
      stubs: { 'se-button': SeButton }
    }
    const wrapper = mount(SeInput, data)
    expect(wrapper.contains('.left.action')).toBe(true)
    expect(wrapper.contains(SeButton)).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have right action button', () => {
    const data = {
      propsData: { type: 'text', action: true, actionPosition: 'right' },
      slots: { rightAction: actionSlot },
      stubs: { 'se-button': SeButton }
    }
    const wrapper = mount(SeInput, data)
    expect(wrapper.contains('.right.action')).toBe(true)
    expect(wrapper.find(SeButton).exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should update data based on input\'s value', done => {
    const parentWrapper = mount(ParentComponent, {
      stubs: { 'se-input': SeInput }
    })
    const seInput = parentWrapper.find(SeInput)
    const newValue = 'helloworld'
    parentWrapper.setData({testdata: 'helloworld'})
    expect(parentWrapper.html()).toMatchSnapshot()
    expect(parentWrapper.contains(SeInput)).toBe(true)
    Vue.nextTick(() => {
      expect(parentWrapper.vm.testdata).toBe(newValue)
      expect(seInput.find('input').element.value).toBe(newValue)
      done()
    })
  })

  it('should update data based event value', done => {
    const parentWrapper = mount(ParentComponent, {
      stubs: { 'se-input': SeInput }
    })
    const seInput = parentWrapper.find(SeInput)
    const newValue = 'helloworld'
    seInput.vm.$emit('input', newValue)
    expect(parentWrapper.html()).toMatchSnapshot()
    expect(parentWrapper.contains(SeInput)).toBe(true)
    Vue.nextTick(() => {
      expect(parentWrapper.vm.testdata).toBe(newValue)
      done()
    })
  })
})
