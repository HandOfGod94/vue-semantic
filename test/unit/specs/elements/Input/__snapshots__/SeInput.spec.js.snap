// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeInput props testing should have focus on the filed 1`] = `
<div class="ui focus input">
    <input type="text">
    <!---->
</div>
`;

exports[`SeInput props testing should have left action button 1`] = `
<div class="ui left action input">
    <button class="ui teal  labeled icon button">
        <i class="cart icon"></i>Checkout</button>
    <input type="text">
    <!---->
</div>
`;

exports[`SeInput props testing should have right action button 1`] = `
<div class="ui right action input">
    <input type="text">
    <!---->
    <button class="ui teal  labeled icon button">
        <i class="cart icon"></i>Checkout</button>
</div>
`;

exports[`SeInput props testing should render basic text input field 1`] = `
<div class="ui input">
    <input type="text">
    <!---->
</div>
`;

exports[`SeInput props testing should update data based event value 1`] = `
<div id="test">
    <div class="ui input">
        <input type="text" name="test">
        <!---->
    </div>
</div>
`;

exports[`SeInput props testing should update data based on input's value 1`] = `
<div id="test">
    <div class="ui input">
        <input type="text" name="test">
        <!---->
    </div>
</div>
`;
